Angular InstantSearch/conditional-request/package.json

```diff
@@ -1,52 +1,42 @@
 {
   "name": "conditional-request",
   "version": "1.0.0",
-  "private": true,
   "scripts": {
     "ng": "ng",
     "start": "ng serve --port 3000",
-    "build": "ng build --prod --progress=false",
-    "test": "ng test",
-    "lint": "ng lint"
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
   },
+  "private": true,
   "dependencies": {
-    "@angular/animations": "6.1.10",
-    "@angular/common": "6.1.10",
-    "@angular/compiler": "6.1.10",
-    "@angular/core": "6.1.10",
-    "@angular/forms": "6.1.10",
-    "@angular/http": "6.1.10",
-    "@angular/platform-browser": "6.1.10",
-    "@angular/platform-browser-dynamic": "6.1.10",
-    "@angular/router": "6.1.10",
-    "@trademe/ng-defer-load": "3.0.1",
-    "algoliasearch": "3.33.0",
-    "angular-instantsearch": "3.0.0-beta.2",
-    "instantsearch.js": "3.7.0",
-    "core-js": "2.6.7",
-    "rxjs": "6.3.3",
-    "zone.js": "0.9.1"
+    "@angular/animations": "~12.2.0",
+    "@angular/common": "~12.2.0",
+    "@angular/compiler": "~12.2.0",
+    "@angular/core": "~12.2.0",
+    "@angular/forms": "~12.2.0",
+    "@angular/platform-browser": "~12.2.0",
+    "@angular/platform-browser-dynamic": "~12.2.0",
+    "@angular/router": "~12.2.0",
+    "algoliasearch": "^4.10.5",
+    "angular-instantsearch": "4.0.0",
+    "rxjs": "~6.6.0",
+    "tslib": "^2.3.0",
+    "zone.js": "~0.11.4"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "0.13.8",
-    "@angular/cli": "7.3.8",
-    "@angular/compiler-cli": "6.1.10",
-    "@angular/language-service": "6.1.10",
-    "@types/algoliasearch": "3.30.13",
-    "@types/jasmine": "3.3.12",
-    "@types/jasminewd2": "2.0.6",
-    "@types/node": "8.10.47",
-    "codelyzer": "5.1.0",
-    "jasmine-core": "2.99.1",
-    "jasmine-spec-reporter": "4.2.1",
-    "karma": "4.1.0",
-    "karma-chrome-launcher": "2.2.0",
-    "karma-coverage-istanbul-reporter": "2.0.5",
-    "karma-jasmine": "2.0.1",
-    "karma-jasmine-html-reporter": "0.2.2",
-    "prettier": "1.17.1",
-    "ts-node": "8.1.0",
-    "tslint": "5.17.0",
-    "typescript": "2.8.4"
+    "@angular-devkit/build-angular": "~12.2.3",
+    "@angular/cli": "~12.2.3",
+    "@angular/compiler-cli": "~12.2.0",
+    "@types/algoliasearch": "^4.0.0",
+    "@types/jasmine": "~3.8.0",
+    "@types/node": "^12.11.1",
+    "jasmine-core": "~3.8.0",
+    "karma": "~6.3.0",
+    "karma-chrome-launcher": "~3.1.0",
+    "karma-coverage": "~2.0.3",
+    "karma-jasmine": "~4.0.0",
+    "karma-jasmine-html-reporter": "~1.7.0",
+    "typescript": "~4.3.5"
```

Comment:
We should pin dependencies.

Reply:
What's nice about not pinning is you don't get as many updates with dependabot. 
The ranges are generated by Angular's CLI so I want to trust them not to break if we happen to update the package.json.